ClusterTest/src/main/scala/net/snowflake/spark/snowflake/BaseTestResultBuilder.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2015-2020 Snowflake Computing
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.snowflake.spark.snowflake
+
+abstract class BaseTestResultBuilder() {
```

Comment:
1. "trait" vs "abstract class". Which should be better in this case.
2. Suppose we should put all new classes as private[snowflake]

Reply:
I've added the package-private modifiers to these classes.

I don't think there's a clear reason yet to use trait vs. class, since we don't expect `BaseTestResultBuilder` to be mixed-in with other types. Classes are slightly more efficient so I'll keep using them here.